addons/hr_holidays/models/hr_leave_accrual_plan_level.py

```diff
@@ -153,90 +174,49 @@ def _compute_level(self):
-                else:
-                    accrual_start = datetime.date(today.year, today.month, accrual.first_day)
-                    accrual_stop = datetime.date(today.year, today.month, accrual.first_day) + relativedelta(months=1)
-                    nextcall = datetime.date(today.year, today.month, accrual.first_day) + relativedelta(months=2)
-            elif frequency == 'biyearly':
-                first_month = MONTHS.index(accrual.first_month) + 1
-                second_month = MONTHS.index(accrual.second_month) + 1
-                potential_first_accrual_date = datetime.date(today.year, first_month, accrual.first_month_day)
-                potential_second_accrual_date = datetime.date(today.year, second_month, accrual.second_month_day)
-                if today <= potential_first_accrual_date:
-                    accrual_start = potential_second_accrual_date - relativedelta(years=1)
-                    accrual_stop = potential_first_accrual_date
-                    nextcall = potential_second_accrual_date
-                else:
-                    if today <= potential_second_accrual_date:
-                        accrual_start = potential_first_accrual_date
-                        accrual_stop = potential_second_accrual_date
-                        nextcall = potential_first_accrual_date + relativedelta(years=1)
-                    else:
-                        accrual_start = potential_second_accrual_date
-                        accrual_stop = potential_first_accrual_date + relativedelta(years=1)
-                        nextcall = potential_first_accrual_date + relativedelta(years=1)
-            elif frequency == 'yearly':
-                month = MONTHS.index(accrual.yearly_month) + 1
-                potential_accrual_date = datetime.date(today.year, month, accrual.yearly_day)
-                if today <= potential_accrual_date:
-                    accrual_start = potential_accrual_date - relativedelta(years=1)
-                    accrual_stop = potential_accrual_date
-                    nextcall = potential_accrual_date + relativedelta(years=1)
-                else:
-                    accrual_start = potential_accrual_date
-                    accrual_stop = potential_accrual_date + relativedelta(years=1)
-                    nextcall = accrual_stop
+    @api.depends('first_day', 'second_day', 'first_month_day', 'second_month_day', 'yearly_day')
+    def _compute_days_display(self):
+        for level in self:
+            level.first_day_display = DAY_SELECT[min(level.first_day - 1, 28)][0]
+            level.second_day_display = DAY_SELECT[min(level.second_day - 1, 28)][0]
+            level.first_month_day_display = DAY_SELECT[min(level.first_month_day - 1, 28)][0]
+            level.second_month_day_display = DAY_SELECT[min(level.second_month_day - 1, 28)][0]
+            level.yearly_day_display = DAY_SELECT[min(level.yearly_day - 1, 28)][0]
+
+    def _inverse_first_day_display(self):
```

Comment:
Have to say this feels hackish (and lot of copy-paste).

Reply:
Ideally we should keep only two of those 'day' fields, but it can not be done in stable unfortunately

Reply:
I think a task should be created to cleanup the bad code that was merged due to the rush